src/pipeline.runner.ts

```diff
@@ -127,6 +128,13 @@ export class PipelineRunner {
 
         log.LogPipelineObject(releaseDefinition);
 
+        // Create ConfigurationVariableValue objects from the input variables
+        let variables = JSON.parse(this.taskParameters.azurePipelineVariables)

```

Comment:
This will cause an unhandled exception when `this.taskParameters.azurePipelineVariables` is null or undefined. As this is an optional param best to wrap this in a conditional, e.g.:
```
        let variables = undefined;
        if (this.taskParameters.azurePipelineVariables) {
            variables = JSON.parse(this.taskParameters.azurePipelineVariables);
            Object.keys(variables).map(function (key, index) {
                let oldValue = variables[key]
                variables[key] = { value: oldValue }
            });
        }
```

Reply:
Thanks, @JamesBurnside. I rebased on the upstream and preferred to squash/fpush the original changes. I kept your suggestions as a separate commit though.

Reply:
Thanks for making these changes to such an outstanding PR (in both meanings of outstanding 😄). I've pinged the original owners of this action on our internal communication at Microsoft so hopefully we'll see some traction on getting this checking in